save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt

```diff
@@ -103,11 +103,16 @@ class DockerService(private val configProperties: ConfigProperties) {
                 false
             }
 
-    private fun buildBaseImageForExecution(execution: Execution): String {
+    private fun buildBaseImageForExecution(execution: Execution): Pair<String, String> {
         val resourcesPath = File(
             configProperties.testResources.basePath,
             execution.resourcesRootPath,
         )
+        val runCmd = if (File(resourcesPath, "run.sh").exists()) {
+            "./run.sh || ./$SAVE_AGENT_EXECUTABLE_NAME"
```

Comment:
```suggestion
            "./run.sh
```
No need to add fallback behavior, if we are already sure that file exists

Reply:
Fixed

Reply:
You still have this `||` operator, though.